README.md

```diff
@@ -17,3 +15,21 @@ The vizlab wiki at https://github.com/USGS-VIZLAB/vizlab/wiki has help pages on
+  
+  Docker commands all either start with `docker` or `docker-compose`.  They can generally do the same things, but `docker` accepts command-line flags (e.g. `docker run -t tag`) while `docker-compose` uses the `docker-compose.yml` for options.  We mostly use `docker-compose` here so the options can be easily source-controlled.
+
+## To build/work on a viz
+  First, either clone your fork of the repo, or pull down only the `Dockerfile` and `docker-compose.yml`.  In a terminal, go into the directory containing the `Dockerfile` and `docker-compose.yml`, and run `docker-compose build`.  This builds the docker image using the options in `docker-compose.yml`.  Next, run `docker-compose up`.  This creates and starts the docker container and leaves it running, with Rstudio exposed on port 8787.  Go to your web browser and you can log in to Rstudio at `localhost:8787`.  The username is `rstudio`, and password is set in the `docker-compose.yml`.  Now you can use Rstudio as you would normally, i.e. creating a new project from a Github repo.  Note the container does not contain any of your credentials, for Github or elsewhere.  However, the container already contains the DOI root certificate, so HTTPS will work over the network. (Note that you should **not** upload an **image** containing the DOI cert to a public repo.) Files you save in Rstudio will be contained in a docker volume, and will persist beyond the life of the container (unless you delete the volume, of course).  When you are done, log out of Rstudio and run `docker-compose down` in your terminal to stop the docker container.      
+
+##  Package management
+  The Dockerfile has a few Docker-specific commands, but largely consists of shell commands to install packages that you have likely seen before.  The Dockerfile here starts with the [rocker geospatial image](https://hub.docker.com/r/rocker/geospatial), which already has R/Rstudio, geospatial libraries, and many standard packages already installed, so we really only need to add vizlab-specific packages on top of it.  You can go look at the Dockerfiles for the various rocker images to see exactly what is installed and from where.
+  R packages that come installed in the rocker images come from [MRAN's daily CRAN snapshots](https://mran.microsoft.com/documents/rro/reproducibility#snapshots),  corresponding to the date the rocker image was updated.  The `repos` option in R is already set to that same snapshot, so any package you install with `install.packages` without setting a repo will come from the same snapshot.  This obviously isn't the case for packages that are only on GRAN.  If you need specific other package versions, it is probably best to install from Github.  
```

Comment:
This is great. Let's be even more prescriptive and change "This obviously isn't the case for packages that are only on GRAN.  If you need specific other package versions, it is probably best to install from Github.  " to "This isn't the case for GRAN packages, so any non-CRAN packages should be installed from GitHub with a release or commit specified."